internal/auth/oidc/repository_auth_method.go

```diff
@@ -0,0 +1,147 @@
+// and then, after a small amount of time, to the ActivePrivateState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.TestAuthMethod(...) succeeds. No options are currently supported.
+func (r *Repository) MakePrivate(ctx context.Context, authMethodId string, opt ...Option) error {
+	panic("to-do")
+}
+
+// MakePublic will transision an OIDC auth method from either the
+// InactiveState or the ActivePrivateState into the temporary StoppingState
+// and then, after a small amount of time, to the ActivePublicState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.TestAuthMethod(...) succeeds. No options are currently supported.
+func (r *Repository) MakePublic(ctx context.Context, authMethodId string, opt ...Option) error {
+	panic("to-do")
+}
+
+// StartAuth accepts a request to start an OIDC authentication/authorization
+// attempt containing the client_type, the client_version, and a map of client
+// round-tripper k/v pairs.  It returns two URLs.  authUrl is an OIDC
+// authorization request URL.  tokenUrl is the URL the client can use to retrieve
+// the results of the user's OIDC authentication attempt.  No options are
+// currently supported.
+func (r *Repository) StartAuth(ctx context.Context, authMethodId string, clientType, clientVersion string, clientRoundTripKVs map[string]string) (authUrl *url.URL, tokenUrl *url.URL, e error) {
+	panic("to-do")
+}
+
+// ExchangeAuthResponse exchanges a successful OIDC Authentication Response for
+// an ID Token, if the exchange is successful, it returns the Account for the
+// user's ID Token sub claim and the client's final redirect URL. If the
+// exchange fails, it returns nil and an empty string, along with any error
+// that occurred.
+//
+// The steps that typically follow ExchangeAuthResponse in the callback handler
+// should be: iam.LookupUserWithLogin(...) to get the Account's User and then
+// authtoken.CreateAuthToken(...) to create a "pending" token for the
+// successful.
+//
+// For more info on a successful OIDC Authentication Response see:
+// https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse
+//
+// Operation steps:
+//
+// * Decrypt the callbackStateParameter which has been encrypted with the OIDC
+// DEK. If decryption fails, and error is returned. callbackStateParameter
+// decrypted payload includes the token_request_id, nonce and
+// final_redirect_url.
+//
+// * Exchange the callbackCodeParameter for provider tokens and validate the
+// tokens.  Call UserInfo endpoint using access token.
+//
+// * Create/update account using "sub" claim as external ID and setting email
+// and full name for the account.
+//
+// * Return the Account and final redirect URL.
+func (r *Repository) ExchangeAuthResponse(ctx context.Context, authMethodId string, callbackStateParameter string, callbackCodeParameter string) (*Account, string, error) {
```

Comment:
`callbackStateParameter` and `callbackCodeParameter` seem overly verbose

Reply:
Yeah, I'm not in love with the names as well.  Happy to hear suggestions!  I want to make sure there's no ambiguity about what's expected, without having to delve into the godocs if possible.